models/demos/falcon7b/tests/test_perf_falcon.py

```diff
@@ -446,13 +447,23 @@ def test_perf_wh_bare_metal(
         model_location_generator,
         get_tt_cache_path,
         all_devices,
+        async_mode,
     ):
         if num_devices > 1:
             pytest.skip(f"num_devices={num_devices} is not supported on CI yet")
         if model_config_str == "BFLOAT16-L1_SHARDED" and kv_cache_len == 2047:
             pytest.skip(f"kv_cache_len={kv_cache_len} does not fit with L1_SHARDED")
-        devices = get_devices_for_t3000(all_devices, num_devices)
+        if async_mode:
+            if llm_mode == "prefill" and seq_len == 128:
+                pytest.skip(f"Skipping {llm_mode} with {seq_len} in async mode")
+            if llm_mode == "decode" and not (kv_cache_len == 2047):
+                if not (model_config_str == "BFLOAT16-L1_SHARDED" or kv_cache_len == 1024):
+                    pytest.skip(f"Skipping {llm_mode} with {kv_cache_len} in async mode")
```

Comment:
some decode configurations do not work? or bad performance?

Reply:
I'm using this to just test functionality. Didn't want to clog up pipelines by doubling the number of tests, so I just run each model configuration with the longest possible length. Locally tried with all configs and they pass.